ngraph/core/reference/src/runtime/reference/experimental_detectron_proposal_single_image.cpp

```
@@ -0,0 +1,381 @@
+                    bfloat16* scores_ptr = reinterpret_cast<bfloat16*>(pscores);
+                    for (size_t i = 0; i < rois_num; ++i)
+                    {
+                        rois_ptr[4 * i + 0] = bfloat16(output_rois[4 * i + 0]);
+                        rois_ptr[4 * i + 1] = bfloat16(output_rois[4 * i + 1]);
+                        rois_ptr[4 * i + 2] = bfloat16(output_rois[4 * i + 2]);
+                        rois_ptr[4 * i + 3] = bfloat16(output_rois[4 * i + 3]);
+                        scores_ptr[i] = bfloat16(output_scores[i]);
+                    }
+                }
+                break;
+                case element::Type_t::f16:
+                {
+                    float16* rois_ptr = reinterpret_cast<float16*>(prois);
+                    float16* scores_ptr = reinterpret_cast<float16*>(pscores);
+                    for (size_t i = 0; i < rois_num; ++i)
+                    {
+                        rois_ptr[4 * i + 0] = float16(output_rois[4 * i + 0]);
+                        rois_ptr[4 * i + 1] = float16(output_rois[4 * i + 1]);
+                        rois_ptr[4 * i + 2] = float16(output_rois[4 * i + 2]);
+                        rois_ptr[4 * i + 3] = float16(output_rois[4 * i + 3]);
+                        scores_ptr[i] = float16(output_scores[i]);
+                    }
+                }
+                break;
+                case element::Type_t::f32:
+                {
+                    float* rois_ptr = reinterpret_cast<float*>(prois);
+                    float* scores_ptr = reinterpret_cast<float*>(pscores);
+                    memcpy(rois_ptr,
+                           output_rois.data(),
+                           shape_size(output_rois_shape) * sizeof(float));
+                    memcpy(scores_ptr,
+                           output_scores.data(),
+                           shape_size(output_scores_shape) * sizeof(float));
+                }
+                break;
+                default:;
```

Should we throw an exception?

---

Done.